imessage.py

```diff
@@ -80,148 +77,238 @@ class Attachment:
-                   attribute.startswith("file-size"):
-                    segments = attribute.split('-')
-                    if segments[-1].isnumeric():
-                        index = int(segments[-1])
-                        attribute_name = segments[:-1]
-                    else:
-                        index = 0
-                        attribute_name = attribute
-
-                    while index >= len(versions):
-                        versions.append(MMCSFile())
-
-                    val = attrs[attribute_name]
-                    match attribute_name:
-                        case "mmcs-url":
-                            versions[index].url = val
-                        case "mmcs-owner":
-                            versions[index].owner = val
-                        case "mmcs-signature-hex":
-                            versions[index].signature = base64.b16decode(val)
-                        case "file-size":
-                            versions[index].size = int(val)
-                        case "decryption-key":
-                            versions[index].decryption_key = base64.b16decode(val)[1:]
-
-            self.versions = versions
+
+            versions = [InlineFile(b"")]
+
+            print(attrib)
+            # for attribute in attrs:
+            #     if attribute.startswith("mmcs") or \
+            #        attribute.startswith("decryption-key") or \
+            #        attribute.startswith("file-size"):
+            #         segments = attribute.split('-')
+            #         if segments[-1].isnumeric():
+            #             index = int(segments[-1])
+            #             attribute_name = segments[:-1]
+            #         else:
+            #             index = 0
+            #             attribute_name = attribute
+
+            #         while index >= len(versions):
+            #             versions.append(MMCSFile())
+
+            #         val = attrs[attribute_name]
+            #         match attribute_name:
+            #             case "mmcs-url":
+            #                 versions[index].url = val
+            #             case "mmcs-owner":
+            #                 versions[index].owner = val
+            #             case "mmcs-signature-hex":
+            #                 versions[index].signature = base64.b16decode(val)
+            #             case "file-size":
+            #                 versions[index].size = int(val)
+            #             case "decryption-key":
+            #                 versions[index].decryption_key = base64.b16decode(val)[1:]
+
+            self.versions = versions # type: ignore
 
     def __repr__(self):
         return f'<Attachment name="{self.name}" type="{self.mime_type}">'
 
+@dataclass
+class Message:
+    text: str
+    sender: str
+    participants: list[str]
+    id: uuid.UUID
+    _raw: dict | None = None
+    _compressed: bool = True
+    xml: str | None = None
```

Comment:
it could be useful to copy over the doc strings for these variables (since they seem to be removed in this pr)

Reply:
ooops, yes, that is from when I refactored for SMS forwarding